compact_str/src/features/sqlx.rs

```diff
@@ -0,0 +1,97 @@
+use sqlx::database::{
+    HasArguments,
+    HasValueRef,
+};
+use sqlx::encode::IsNull;
+use sqlx::error::BoxDynError;
+use sqlx::{
+    Database,
+    Decode,
+    Encode,
+    Type,
+    Value,
+    ValueRef,
+};
+
+use crate::{
+    CompactString,
+    ToCompactString,
+};
+
+#[cfg_attr(docsrs, doc(cfg(feature = "sqlx")))]
+impl<DB> Type<DB> for CompactString
+where
+    DB: Database,
+    for<'x> &'x str: Type<DB>,
+{
+    #[inline]
+    fn type_info() -> <DB as Database>::TypeInfo {
+        <&str as Type<DB>>::type_info()
+    }
+}
+
+#[cfg_attr(docsrs, doc(cfg(feature = "sqlx")))]
+impl<'r, DB> Decode<'r, DB> for CompactString
+where
+    DB: Database,
+    for<'x> <DB as HasValueRef<'x>>::ValueRef: ValueRef<'x>,
+    for<'x> &'x str: Decode<'x, DB> + Type<DB>,
+{
+    fn decode(value: <DB as HasValueRef<'r>>::ValueRef) -> Result<Self, BoxDynError> {
+        let value = value.to_owned();
+        let value: &str = value.try_decode()?;
+        Ok(value.try_to_compact_string()?)
+    }
+}
+
+macro_rules! impl_encode_common {
+    ($db:ident) => {
+        #[inline]
+        fn produces(&self) -> Option<<sqlx::$db as Database>::TypeInfo> {
+            <&str as Encode<'_, sqlx::$db>>::produces(&self.as_str())
+        }
+
+        #[inline]
+        fn size_hint(&self) -> usize {
+            <&str as Encode<'_, sqlx::$db>>::size_hint(&self.as_str())
+        }
+    };
+}
+
+macro_rules! impl_encode {
+    ($feature:literal => $db:ident) => {
+        #[cfg(feature = $feature)]
+        #[cfg_attr(docsrs, doc(cfg(feature = $feature)))]
+        impl<'q> Encode<'q, sqlx::$db> for CompactString {
+            #[inline]
+            fn encode_by_ref(
+                &self,
+                buf: &mut <sqlx::$db as HasArguments<'q>>::ArgumentBuffer,
+            ) -> IsNull {
+                Encode::<'_, sqlx::$db>::encode_by_ref(&self.as_str(), buf)
+            }
+
+            impl_encode_common!($db);
+        }
+    };
+}
+
+impl_encode!("sqlx-mysql" => MySql);
+impl_encode!("sqlx-postgres" => Postgres);
```

Comment:
Do you mind expanding these macros? Normally I would totally agree that using a macro here and adhering to DRY is the correct approach. But given we're not de-duplicating that much code, I don't believe the added complexity/overhead of using a macro with worth it